src/Features/Core/Portable/GenerateType/AbstractGenerateTypeService.Editor.cs

```diff
@@ -62,6 +62,9 @@ private partial class Editor
                 GenerateTypeOptionsResult generateTypeOptionsResult,
                 CancellationToken cancellationToken)
             {
+                // the document comes from the same snapshot as the project
+                Contract.ThrowIfFalse(document.Project.Solution == generateTypeOptionsResult.Project.Solution);
```

Comment:
What prompted this? Not obvious to me how this ever wasn't true before?

Reply:
This not being true caused the deadlock in the tests. There is a condition in the source of this that compares project snapshots for equality and adds a new project reference. The tests did pass in document from one snapshot and generateTypeOptionsResult from another one causing the condition to fail and add a self-referencing project reference.